server/server.js

```diff
@@ -18,20 +18,27 @@ if (process.env.LOG_LEVEL) {
 }
 
 // schema
-const buildSchema = require('./buildSchema')
+const { Core } = require('@aerogear/data-sync-gql-core')
 const schemaListenerCreator = require('./lib/schemaListeners/schemaListenerCreator')
+const { makeExecutableSchema } = require('graphql-tools')
 
 class DataSyncServer {
-  constructor (config, models, pubsub) {
+  constructor (config, pubsub) {
     this.config = config
-    this.models = models
     this.pubsub = pubsub
 
     this.schema = null
     this.dataSources = null
   }
 
+  async initializeCore (config) {
+    this.core = new Core(config.postgresConfig, makeExecutableSchema)
+    this.models = await this.core.getModels()
+    this.models.sync({ logging: false })
```

Comment:
@StephenCoady Just wondering, what's the rationale behind having this here? This is essentially like running a database migration every time the server starts up.

In master, we **purposefully avoid** calling this and it is up to the user to do initialize the database structure with `npm run db:init`. There are a couple of reasons for this:

* In local development, you might make manual changes to the db structure for whatever reason (new feature, debugging etc.) and this automatic sync would destroy those changes
* Without the automatic sync, it becomes very clear if there is any mismatch between the server and the db structure because the whole thing will crash.

Is there any specific reason this was needed here?

Reply:
But this is only for metadata. Users should not actually manage our metadata storage.
Sync will just create required columns but our metadata format seems to be static.

Reply:
This was an oversight on my part. Originally the server handled the sync of models at the service level before the server was started. I moved it but have moved it back now, will push shortly.